task4/src/components/modal/modal.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import './modal.scss';
+import PropTypes from 'prop-types';
+
+function Modal(props) {
+  return <>
+    <div className="modal-container">
+      <div className="back-shadow"/>
+      <div className={`modal-body ${props.className || ''}`}>
+        <div className="modal-header">
+          <button onClick={props.close}>X</button>
+        </div>
+        {props.children}
+      </div>
+    </div>
+  </>
+}
+
+Modal.protoTypes = {
+  children: PropTypes.node.isRequired,
+  close: PropTypes.func.isRequired,
+  className: PropTypes.string,
+}
+
+export default Modal;
```

Comment:
There are some developers that a default export is not a good practice, that you have to make a named export, but is up to you.

Reply:
ack, will refactor on next PR